### ISSUE TITLE
PrettyTableViewCell cannot be released.

### ISSUE THREAD

**Issue by Jo-H** [vicpenap/PrettyKit#7]
Thanks for the great library!!

I have one problem about memory management!

As title, when I use PrettyTableViewCell, it cannot be released.

I guess it's because of "[self.contentView addObserver:self forKeyPath:@"frame" options:NSKeyValueObservingOptionOld context:nil];"
The "[self.contentView removeObserver:self forKeyPath:@"frame"];" is in "dealloc" method, but without removing the cell cannot be dealloc.

Not sure if I was right. 

Please check this.

Thanks a lot.

**Comment by Jo-H** [comment]
Sorry, I was wrong for the reason.

if I comment the following code, the cell can be released.

---

```
    PrettyTableViewCellBackground *bg = [[PrettyTableViewCellBackground alloc] initWithFrame:self.frame 
                                                                              behavior:CellBackgroundBehaviorNormal];
    bg.cell = self;
    self.backgroundView = bg;
    [bg release];

    bg = [[PrettyTableViewCellBackground alloc] initWithFrame:self.frame
                                                  behavior:CellBackgroundBehaviorSelected];
    bg.cell = self;
    self.selectedBackgroundView = bg;
    [bg release];
```

---

the backgroundview and cell both retain each other, so cannot be released.
I think that's the real reason.

Please Check this.

Thanks a lot!

**Comment by Jo-H** [comment]
I modified the cellBackground property from "retain" to "assign". And it works.
Wonder if somehow the cell is released but the backgroundview ask something from cell and got crash.
Dont know if this would happen.

---

@interface PrettyTableViewCellBackground : UIView

//@property (nonatomic, retain) PrettyTableViewCell *cell;
@property (nonatomic, assign) PrettyTableViewCell *cell;
@property (nonatomic, assign) CellBackgroundBehavior behavior;

---

**Comment by vicpenap** [comment]
Thanks!

You're right about that. It shouldn't crash though, as the background is only used from the cell, so an assign there should do the trick. I'll use you as a tester ;). Tell me if everything keeps working OK after a week or so :)

Again, thanks!

**Comment by vicpenap** [comment]
Everything keeps working OK?

**Comment by Jo-H** [comment]
So far it' s ok.

But not tested heavily...

Thanks!

**Comment by vicpenap** [comment]
Fixed in commit 52366d5d719b. Will be included in the next release.
